### PR TITLE
[WIP] :: AI :: refactor loading logic

### PR DESCRIPTION
**Pull request by n3f**

### CODE REVIEW

**Comment by n3f** [comment]
@millerf, I was getting lost in all the is loading stuff -- I took a stab at refactoring --  I haven't tested it but wanted to get your thoughts.

**Comment by millerf** [comment]
@n3f I almost missed this PR... Looking now...

**Comment by millerf** [comment]
It does not work at all :)

https://user-images.githubusercontent.com/790558/224287040-d02a90d9-be83-4699-b779-1770d5cf4baa.mov

File: projects/plugins/jetpack/extensions/blocks/ai-paragraph/edit.js [hunk 1]
```diff
@@ -158,14 +138,14 @@ export default function Edit( { attributes: { state }, setAttributes, clientId }
...
 	const getSuggestionFromOpenAI = useCallback( () => {
-		if ( !! content || isLoadingCompletion || is_triggered( state ) ) {
+		if ( !! content || is_triggered( state ) ) {
 			return;
 		}
 
 		setShowRetry( false );
 		setErrorMessage( false );
 		setNeedsMoreCharacters( false );
-		setIsLoadingCompletion( true );
```
Comment (millerf) [inline projects/plugins/jetpack/extensions/blocks/ai-paragraph/edit.js:196]: So you replaced the isLoadingCompletion with your state. Makes sense 👍

File: projects/plugins/jetpack/extensions/blocks/ai-paragraph/edit.js [hunk 2]
```diff
@@ -72,76 +73,55 @@ export const createPrompt = (
...
+	const [ categoryObjects, categoriesLoaded ] = useTaxonomy( 'categories', 'category' );
 	const categoryNames = categoryObjects
 		.filter( cat => cat.id !== 1 )
 		.map( ( { name } ) => name )
 		.join( ', ' );
+
+	const [ tagObjects, tagsLoaded ] = useTaxonomy( 'tags', 'post_tag' );
 	const tagNames = tagObjects.map( ( { name } ) => name ).join( ', ' );
 
+	const isTaxonomyLoaded = categoriesLoaded && tagsLoaded;
```
Comment (millerf) [inline projects/plugins/jetpack/extensions/blocks/ai-paragraph/edit.js]: So `isTaxonomyLoaded` never gets to True. I could not debug more, for some reason PHPStorm keeps on freezing....

But the refactor in itself seems like a very good idea (`isLoadingCompletion` is defintiely superfluous now)...

**Comment by n3f** [comment]
OK @millerf after playing with it more today I understand all the logic much better and I pretty much rolled most of it back.  I did some rudimentary testing and it was working for me (but I imagine there might still be 1 or 2 issues with the errorMessage edge cases). But I don't know if the changes are even worth it at this point.  Maybe @artpi can chime in (if he cares).

I saw Artur's comment on your other PR about not dirtying the attributes (and I totally agree), so I'm wondering if I should even just get rid of the state attribute I added. 🤷

**Pull request closed without merging**